models.py

```diff
@@ -5,8 +5,16 @@ def __init__(self, name, hp=100.0, strength=0.0, defence=1.0):
         self.strength = strength
         self.defence = defence
 
+    def attack_strenght(self,strength):
```

Comment:
If You want to implement this function it in the `Character` class (like You did), You don't need to pass the 'strength' parameter, because You have access to the character's strength by `self.strength`.

I'd rather name this method 'generate_random_damage', it would take no additional arguments (only `self`) and return generated damage with the random factor.
Sth like:
```
def generate_random_damage(self):
    (magic happens here)
    return generated_damage
```
then we should be able to use it like this:
`damage = self.generate_random_damage() - opponent.defence`
sexy approach, isn't it?